annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+
+
+def test_create_tasks(mock_session: Mock):
+    tasks = [
+        {"user_id": 1, "file_id": 123, "pages": {1, 2}},
+        {"user_id": 2, "file_id": 456, "pages": {3}},
+    ]
+    job_id = 1
+    expected_user_ids = {1, 2}
+    with patch(
+        "annotation.tasks.services.update_files"
+    ) as mock_update_files, patch(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+        mock_bulk_insert_mappings = MagicMock()
+        mock_session.bulk_insert_mappings = mock_bulk_insert_mappings
+        mock_session.flush = MagicMock()
+
+        create_tasks(mock_session, tasks, job_id)
+
+        assert mock_bulk_insert_mappings.call_count == 1
+        assert mock_update_files.call_count == 1
+
+        mock_update_user_overall_load.assert_has_calls(
+            [call(mock_session, user_id) for user_id in expected_user_ids],
+            any_order=True,
+        )
+
+
+def test_create_tasks_with_empty_tasks(mock_session: Mock):
+    tasks = []
+    job_id = 1
+    with patch(
+        "annotation.tasks.services.update_files"
+    ) as mock_update_files, patch(
+        "annotation.tasks.services.update_user_overall_load"
+    ) as mock_update_user_overall_load:
+        mock_bulk_insert_mappings = MagicMock()
+        mock_session.bulk_insert_mappings = mock_bulk_insert_mappings
+        mock_session.flush = MagicMock()
+
+        create_tasks(mock_session, tasks, job_id)
+
+        assert mock_bulk_insert_mappings.call_count == 1
+        assert mock_update_files.call_count == 1
+        mock_update_user_overall_load.assert_not_called()
+
+
+def create_task(status: TaskStatusEnumSchema) -> ManualAnnotationTask:
+    task = MagicMock(spec=ManualAnnotationTask)
+    task.status = status
+    task.id = 1
+    return task
+
+
+def test_update_task_status_ready(mock_session: Mock):
+    task = create_task(TaskStatusEnumSchema.ready)
+
+    update_task_status(mock_session, task)
+
+    assert task.status == TaskStatusEnumSchema.in_progress
+    mock_session.add.assert_called_once_with(task)
+    mock_session.commit.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("status", "expected_message"),
+    (
+        (TaskStatusEnumSchema.pending, "Job is not started yet"),
+        (TaskStatusEnumSchema.finished, "Task is already finished"),
+    ),
+)
+def test_update_task_status_error(
+    mock_session: Mock, status: TaskStatusEnumSchema, expected_message: str
+):
+    task = create_task(status)
+    with pytest.raises(FieldConstraintError, match=f".*{expected_message}.*"):
+        update_task_status(mock_session, task)
+
+
+def test_finish_validation_task(mock_session: MagicMock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.with_for_update.return_value = mock_query
+    mock_query.update.return_value = None
```

Comment:
Seems too complicated, as you already have a mock_session.

Reply:
I wrote it in one line, but flake8 doesn't allow the commit.

Reply:
```suggestion
    mock_session.update.return_value = None
```
Did you write it like this?

Reply:
It prints an error: AttributeError: Mock object has no attribute 'update'